etcdserver/api/v3rpc/interceptor.go

```diff
@@ -84,6 +84,16 @@ func newLogUnaryInterceptor(s *etcdserver.EtcdServer) grpc.UnaryServerIntercepto
 
 func logUnaryRequestStats(ctx context.Context, lg *zap.Logger, info *grpc.UnaryServerInfo, startTime time.Time, req interface{}, resp interface{}) {
 	duration := time.Since(startTime)
+	var enabledDebugLevel, expensiveRequest bool
+	if lg.Core().Enabled(zap.DebugLevel) {
+		enabledDebugLevel = true
+	}
+	if duration > etcdserver.WarnApplyDuration {
```

Comment:
I am thinking maybe we want to define a separate interval for this? The handling chain includes many steps besides `apply step`.

Reply:
yes. 100ms looks smaller,so here i want to define 300ms?  I expect it will log some very expensive requests, which need to be paid attention to in time, try not to duplicate with the warn log. how do you think so? thanks. @jingyih

Reply:
300ms sounds good. When this log line is triggered and printed, is there a way to correlate it to the corresponding trace prints (so that we know which step takes more time)?

Reply:
if we want to get trace info, we can enable trace features and add trace logs for more RPC methods in the future. the log has the following two purposes. On the one hand, it can tell us expensive request's remote info, it is useful for non-k8s business(There are many clients,no apiserver), on the other hand, it can print some warning logs for rpc method such as Authenticate. There was a troubleshooting case before, the user reported that the range request timeout exceeded 5S, and there was no log. finally, I find it was caused by the authentication, not the range request itself.  updated. PTAL. @jingyih